inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp

```diff
@@ -174,18 +175,28 @@ ie::CNNNetwork FrontEnd::convertNetwork(ie::CNNNetwork& network) {
     manager.register_pass<vpu::MergeGatherGatherElements>();
     manager.register_pass<ngraph::pass::CommonOptimizations>();
 
-    manager.register_pass<vpu::ExtractBatch>(std::unordered_set<ngraph::Node::type_info_t>{
+    manager.register_pass<vpu::ExtractBatch>(std::unordered_set<ngraph::Node::type_info_t> {
         ngraph::opset5::MatMul::type_info,
         ngraph::opset5::Convolution::type_info,
         ngraph::opset5::GroupConvolution::type_info
     });
-
     manager.register_pass<vpu::DynamicToStaticShape>();
     manager.register_pass<vpu::EliminateShapeOfAfterDSR>();
     manager.register_pass<vpu::ConvertExtractImagePatchesToReorgYolo>();
     manager.register_pass<ngraph::pass::ConvertOpSet3ToOpSet2>();
     manager.register_pass<ngraph::pass::ConvertOpSet2ToOpSet1>();
+    // ConvertPrecision must be executed before ConvertOpSet1ToLegacy due to this pass works with operations from opsets only
+    manager.register_pass<ngraph::pass::ConvertPrecision>(ngraph::element::i64, ngraph::element::i32, myriadTypeToFuseMap);
+    manager.register_pass<ngraph::pass::ConvertPrecision>(ngraph::element::u64, ngraph::element::i32, myriadTypeToFuseMap);
+    manager.register_pass<ngraph::pass::ConvertPrecision>(ngraph::element::u32, ngraph::element::i32, myriadTypeToFuseMap);
+
     manager.register_pass<ngraph::pass::ConvertOpSet1ToLegacy>();
+    // Convert BOOL->I64 must be executed after ConstantFolding to avoid precision type missmatch when we try to evaluate nodes with BOOL output.
+    // For example evaluate_greater_equal calls set_broadcast function with hardcoded BOOL precision.
+    // In set_broadcast function we compare original node's precision with hardcoded so we get an error if we change precision before.
+    manager.register_pass<ngraph::pass::ConvertPrecision>(ngraph::element::boolean, ngraph::element::i32, myriadTypeToFuseMap);
```

Comment:
`ConvertOpSet1ToLegacy ` potentially can do smth with operation that produces boolean type so `ConvertPrecision` can miss new operation types.
I suggest to call ConstantFolding explicitly in this pipeline. I think we should keep it somewhere after `DynamicToStaticShape` and then register boolean precision converter before legacy conversion.

Reply:
Fixed.

Reply:
Could you please explain how Constant Folding helps with the fact ConvertOpset1ToLegacy may insert new operations producing boolean?

Reply:
`ConvertOpset1ToLegacy` do not produce new ops with boolean type. And in worst case it will add i64 Constants. So that's why we add i64->i32 conversion in CPU and GPU plugins.

Reply:
So we insert ConstantFolding to fold those constants if possible and only in other cases insert convertions?

Reply:
As I understand, we have the following situation:
CF trying to evaluate nodes with fully constant inputs and such nodes are then replaced with new constants containing the result of a folded operation. 
We called CF in CommonOptimization and in ConvertOpSet1ToLegacy pass(maybe more, but it doesn't matter).
But now we need to call CF explicit after DTS. These transformations can produce nodes, which can be folded.
If we didn't do it we can catch an error in ConvertOpSet1ToLegacy pass due to some evaluators wait for nodes with specific precision (for example evaluate_greater_equal calls set_broadcast function with hardcoded BOOL precision) but we change the precision before.

Reply:
Ok, thanks for explanation